src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { getUser } from '../../api';
+
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then((u) => {
+        setUser(u);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        throw new Error('Error fetching user: ', error.message);
+      });
+  }, []);
```

Comment:
Missing dependency in useEffect, add todo.userId to dependencies array to avoid unexpected behavior.